src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,79 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String COMMA = ",";
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+    private int supplyAmount = 0;
+    private int buyAmount = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        supplyAmount = 0;
+        buyAmount = 0;
+        File fromFile = new File(fromFileName);
+        final File toFile = new File(toFileName);
+
+        calculateReport(fromFile, stringBuilder);
+
+        writeIntoFile(toFile, stringBuilder);
+    }
+
+    private void calculateReport(File fromFile, StringBuilder stringBuilder) {
+
+        readFromFile(fromFile);
+
+        stringBuilder.append(SUPPLY).append(COMMA).append(supplyAmount)
+                .append(System.lineSeparator())
+                .append(BUY).append(COMMA).append(buyAmount)
+                .append(System.lineSeparator())
+                .append(RESULT).append(COMMA).append(supplyAmount - buyAmount);
```

Comment:
The `calculateReport` method is doing two things: reading from the file and building a string for the report. This violates the principle that a method should be responsible for one task. Consider splitting this method into two: one for reading the file and calculating amounts, and another for building the report string.